testing/end-to-end/tests/base/factory-reset.nix

```diff
@@ -0,0 +1,43 @@
+{pkgs, disk, overlayPath, ...}:
+pkgs.testers.runNixOSTest {
+  name = "Factory reset works";
+
+  nodes = {
+    playos = { config, lib, pkgs, ... }:
+    {
+      imports = [
+        (import ../../virtualisation-config.nix { inherit overlayPath; })
+      ];
+    };
+  };
+
+  extraPythonPackages = ps: [
+    ps.colorama
+    ps.types-colorama
+  ];
+
+  testScript = ''
+${builtins.readFile ../../test-script-helpers.py}
+create_overlay("${disk}", "${overlayPath}")
+
+playos.start(allow_reboot=True)
+
+with TestPrecondition("Persistent data is mounted"):
+    playos.wait_for_unit('mnt-data.mount')
+
+with TestCase("Persistent data remains after reboot"):
+    playos.succeed("echo TEST_DATA > /mnt/data/persist-me")
+    playos.shutdown()
+    playos.start(allow_reboot=True)
+    playos.wait_for_unit('mnt-data.mount')
+    playos.succeed("grep TEST_DATA /mnt/data/persist-me")
+
+with TestCase("Persistent data is wiped if factory reset is triggered"):
+    playos.succeed("systemctl start playos-wipe-persistent-data.service")
```

Comment:
I guess doing this via an actual HTTP request must fail somehow, or could we do this instead?

Reply:
This absolutely can be done using an HTTP request, but I on purpose wanted to split the tests into two parts: testing the `playos-wipe-persistent-data.service` only (because it is a very tricky thing, took me ages to get it right!) and testing the REST endpoints of the controller GUI.

Reply:
Simplified in 5c8823880a2